### PR TITLE
Update analyzer, refactor code, and add abstract class support

### LINKED ISSUE #1: Bad constructor parameters does not work for abstract classes
Current analyzer uses fake "new T(args)" to understand whether constructor exists. It does not work for abstract classes because they cannot be instantiated directly.

### PR DESCRIPTION
**Pull request by rjmurillo**
This commit includes several updates and improvements. 

- The `ConstructorArgumentsShouldMatchAnalyzerTests.ShouldFailIfClassParametersDoNotMatch.approved.txt` file has been updated to reflect changes in the source code. 
- The `FooAbstract` class and its usage have been removed from `ConstructorArgumentsShouldMatch.cs`. 
- Defensive programming practices have been added to `DiagnosticVerifier.Helper.cs` to ensure non-null arguments. 
- A new file `AbstractClass.cs` has been added to `Moq.Analyzers.Test.csproj`. 
- The `Analyze` method in `ConstructorArgumentsShouldMatchAnalyzer.cs` has been refactored for better readability and maintainability, and now supports abstract classes. 
- New test class `AbstractClassTests` and related files have been added to verify the functionality of the updated analyzer.

Fixes #1

### CODE REVIEW

**Comment by rjmurillo** [comment]
@MattKotsenas the build action introduced in #9 didn't run on this PR

**Comment by MattKotsenas** [comment]
> @MattKotsenas the build action introduced in #9 didn't run on this PR

Copy/paste error on my part. The default branch is `master` and my code uses `main`.

Will update to master until we have the ability to change the default branch.

**Comment by MattKotsenas** [comment]
Actions issue is fixed in #21.

**Comment by rjmurillo** [comment]
This is failing between platforms. On Windows and through Ubuntu in WSL, the line numbers are a certain value. On a stand alone Ubuntu VM, they're about 50 lines off. 

```diff
Diagnostic 2
	Id: Moq1300
-	Location: SourceFile(Test0.cs[1594..1604))
+	Location: SourceFile(Test0.cs[1534..1544))
	Highlight: OtherClass
	Lines: mock.As<OtherClass>();
	Severity: Error
	Message: Mock.As() should take interfaces only
```

Still looking into this

File: Source/Moq.Analyzers.Test/Data/AbstractClass.Bad.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Moq.Analyzers.Test.Data
+{
+    internal class MyBadUnitTests
+    {
+        private void TestBad()
```
Comment (MattKotsenas) [inline Source/Moq.Analyzers.Test/Data/AbstractClass.Bad.cs:5]: From conversation, some test cases missing:

- Pass an argument to a constructor that takes no arguments
- Pass (string, int) to a constructor that takes (int, string) or equivalent test
- Pass the incorrect number of arguments to constructor

**Review by MattKotsenas — CHANGES_REQUESTED** [review]

**Review by MattKotsenas — APPROVED** [review]

**Pull request merged**